src/csvcubedpmd/csvwcli/pull.py

```diff
@@ -159,20 +164,21 @@ def pull(csvw_metadata_url: str, output_dir: Path) -> None:
 
         output_file = output_dir / relative_dependency_path
         _ensure_dir_structure_exists(output_file.parent)
-        _download_to_file(absolute_dependency_path, output_file)
+        if _looks_like_uri(csvw_metadata_url):
+            _download_to_file(absolute_dependency_path, output_file)
+        else:
+            shutil.copy(absolute_dependency_path, output_file)
 
 
 def _get_file_name_from_url(url: str) -> str:
     return os.path.basename(urlparse(url).path)
 
-
 def _download_to_file(rel_dep_url: str, output_file: Path) -> None:
     with open(output_file, "wb+") as f:
         response = requests.get(rel_dep_url)
         for chunk in response.iter_content(chunk_size=1024):
             f.write(chunk)
 
-
 def _ensure_dir_structure_exists(dir_path: Path) -> None:
     if not dir_path.exists():
```

Comment:
90% certain we can do the following and avoid the recursive logic

```python
def _ensure_dir_structure_exists(dir_path: Path) -> None:
     dir_path.mkdir(exist_ok=True, parents=True)
```

just ignore if I'm wrong but I don't think I am.

Reply:
I think you're right @mikeAdamss , I wrote this bit of code before I knew of said param.